packages/unmock-core/src/service/service.ts

```diff
@@ -1,37 +1,123 @@
-import { filter as _filter } from "lodash";
-import { HTTPMethod, IUnmockServiceState, OASSchema } from "./interfaces";
+import XRegExp from "xregexp";
+import { getAtLevel } from "../util";
+import {
+  DEFAULT_ENDPOINT,
+  DEFAULT_REST_METHOD,
+  OAS_PARAMS_KW,
+  OAS_PATH_PARAM_REGEXP,
+  UNMOCK_PATH_REGEX_KW,
+} from "./constants";
+import {
+  HTTPMethod,
+  IService,
+  IUnmockServiceState,
+  OASSchema,
+} from "./interfaces";
 
 /**
  * Implements the state management for a service
  */
 
 // TODO: Is there room here for xstate, in the future?
-// TODO: Add IService
-export class Service {
+
+export class Service implements IService {
   // Maintains a state for service
   // First level kv pairs: paths -> methods
   // Second level kv pairs: methods -> status codes
   // Third level kv pairs: status codes -> response template overrides
   // Fourth and beyond: template-specific.
-  // @ts-ignore
+  // @ts-ignore // ignored because it's currently only being read and not written
```

Comment:
Doesn't our TS config allow that :o Seems to me more like a thing linter would complain about.

Reply:
It doesn't. I didn't want to take matters into my own hands as the state is temporarily not being used. Seems like a good protection IMO, linting-wise.